integration-tests/src/mock_network/setup.rs

```diff
@@ -0,0 +1,177 @@
+    let path = get_store_path(home_dir);
+    let store = create_store(&path);
+
+    Arc::new(NightshadeRuntime::with_config(
+        home_dir,
+        store.clone(),
+        config,
+        config.client_config.trie_viewer_state_size_limit,
+        config.client_config.max_gas_burnt_view,
+    ))
+}
+
+#[allow(dead_code)]
+fn setup_mock_peer_manager_actor(
+    runtime: Arc<NightshadeRuntime>,
+    client_addr: Recipient<NetworkClientMessages>,
+    genesis_config: &GenesisConfig,
+    chain_genesis: &ChainGenesis,
+    block_production_delay: Duration,
+) -> MockPeerManagerActor {
+    let chain =
+        Chain::new_for_view_client(runtime, chain_genesis, DoomslugThresholdMode::NoApprovals)
+            .unwrap();
+    MockPeerManagerActor::new(client_addr, genesis_config, chain, 0, block_production_delay)
+}
+
+#[allow(dead_code)]
+fn setup_mock_network(
+    client_home_dir: &Path,
+    network_home_dir: &Path,
+    config: &NearConfig,
+) -> (Addr<MockPeerManagerActor>, Addr<ClientActor>, Addr<ViewClientActor>) {
+    let client_runtime = setup_runtime(client_home_dir, &config);
+    let mock_network_runtime = setup_runtime(network_home_dir, &config);
+
+    let telemetry = TelemetryActor::new(config.telemetry_config.clone()).start();
+    let chain_genesis = ChainGenesis::from(&config.genesis);
+
+    let node_id = PeerId::new(config.network_config.public_key.clone().into());
+    let network_adapter = Arc::new(NetworkRecipient::default());
+    #[cfg(feature = "test_features")]
+    let adv = Arc::new(std::sync::RwLock::new(AdversarialControls::default()));
+
+    let block_production_delay = config.client_config.min_block_production_delay;
+    let (client_actor, _) = start_client(
+        config.client_config.clone(),
+        chain_genesis.clone(),
+        client_runtime.clone(),
+        node_id,
+        network_adapter.clone(),
+        config.validator_signer.clone(),
+        telemetry,
+        #[cfg(feature = "test_features")]
+        adv.clone(),
+    );
+
+    let view_client = start_view_client(
+        None,
+        chain_genesis.clone(),
+        client_runtime.clone(),
+        network_adapter.clone(),
+        config.client_config.clone(),
+        #[cfg(feature = "test_features")]
+        adv.clone(),
+    );
+
+    let arbiter = Arbiter::new();
+    let client_actor1 = client_actor.clone();
+    let genesis_config = config.genesis.config.clone();
+    let mock_network_actor =
+        MockPeerManagerActor::start_in_arbiter(&arbiter.handle(), move |_ctx| {
+            setup_mock_peer_manager_actor(
+                mock_network_runtime,
+                client_actor1.recipient(),
+                &genesis_config,
+                &chain_genesis,
+                block_production_delay,
+            )
+        });
+    network_adapter.set_recipient(mock_network_actor.clone().recipient());
+    (mock_network_actor, client_actor, view_client)
+}
+
```

Comment:
Great - now we're only missing a binary, where you 'point' it at existing (for example mainnet) .near/ home directory and an 'empty' one -- and then it should start 'syncing' ...

Reply:
added